appyx-interactions/common/src/commonMain/kotlin/com/bumble/appyx/interactions/core/Keyframes.kt

```diff
@@ -85,11 +88,11 @@ data class Keyframes<ModelState>(
         val progress = progress.coerceIn(0f, maxProgress)
         Logger.log("Keyframes", "$progress")
         progressFlow.value = progress
-        segmentProgress.value = progress - currentIndex
         if (progress.toInt() > currentProgress) {
             Logger.log("Keyframes", "onTransitionFinished()")
             onTransitionFinished(currentSegment.fromState)
         }
     }
 }
+internal fun Float.toSegmentProgress(segmentIndex: Int) = this - segmentIndex
```

Comment:
However, with this solution values `> 1f` will be capped at `1f` and cause extra emissions that are conceptually not correct (if it's out of range, it shouldn't trigger anything).

Reply:
Yep, this is exactly why I used filter to stop emitting after reaching 1. I'm not confident if distinctUntilChanged will help in this case because of float comparison